### PR TITLE
James 3354: Mailbox/set creation: parentId handling

### PR DESCRIPTION
**Pull request by nvduc91**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala [hunk 1]
```diff
@@ -393,4 +397,276 @@ trait MailboxSetMethodContract {
...
+      s"""{
+         |  "sessionState": "75128aab4b1b",
+         |  "methodResponses": [[
+         |    "Mailbox/set",
+         |    {
+         |      "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |      "newState": "000001",
+         |      "notCreated": {
+         |        "C42": {
+         |          "type": "invalidArguments",
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:663]: Should be "forbidden"

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxSetMethod.scala [hunk 2]
```diff
@@ -23,34 +23,69 @@ import eu.timepit.refined.auto._
...
+  }
+}
+
+case class CreationResults(created: Seq[CreationResult]) {
+  def retrieveCreated: Map[MailboxCreationId, MailboxId] = created
+    .flatMap(result => result match {
+      case success: CreationSuccess => Some(success.mailboxCreationId, success.mailboxId)
+      case _ => None
+    })
+    .toSet
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxSetMethod.scala:63]: toSet is not needed here

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxSetMethod.scala [hunk 3]
```diff
@@ -23,34 +23,69 @@ import eu.timepit.refined.auto._
...
+    .toMap
+
+  def retrieveErrors: Map[MailboxCreationId, MailboxSetError] = created
+    .flatMap(result => result match {
+      case failure: CreationFailure => Some(failure.mailboxCreationId, failure.asMailboxSetError)
+      case _ => None
+    })
+    .toSet
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxSetMethod.scala:71]: toSet is not needed here

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala [hunk 4]
```diff
@@ -393,4 +397,276 @@ trait MailboxSetMethodContract {
...
+        |}
+        |""".stripMargin
+
+      `given`
+        .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+        .body(request)
+      .when
+        .post
+      .`then`
+        .log().ifValidationFails()
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:432]: question: is this only for debugging or not? I start to see this a bit everywhere in jmap tests

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala [hunk 5]
```diff
@@ -606,4 +607,66 @@ trait MailboxSetMethodContract {
          |    "c1"]]
          |}""".stripMargin)
   }
+
+  @Test
+  def mailboxSetShouldNotCreateMailboxWhenMailboxItShared(server: GuiceJamesServer): Unit = {
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:612]: Name of test is wrong...

How about `mailboxSetShouldNotCreateChildMailboxWhenSharedParentMailbox` ? Not sure it's the best though, opened to better names

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala (hunk 4 above)
Comment (nvduc91) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:432]: Im not sure, but let's keep it that way.

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:432]: > question: is this only for debugging or not?

If the test fails, you are prompted out why, and can easily debug.

If the test succeeds, no boiler plate output.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**